tesseract_ros/tesseract_rosutils/include/tesseract_rosutils/plotting.h

```diff
@@ -72,6 +73,25 @@ class ROSPlotting : public tesseract_visualization::Visualization
     trajectory_pub_.publish(msg);
   }
 
+  /**
+   * @brief Plots waypoints according to their type. Currently only CARTESIAN_WAYPOINT is implemented.
+   * @param waypoints A vector of waypoint pointers.
+   */
+  void plotWaypoints(const std::vector<tesseract_motion_planners::Waypoint::Ptr>& waypoints)
+  {
+    for (auto& waypoint : waypoints)
+    {
+      if (waypoint->getType() == tesseract_motion_planners::WaypointType::CARTESIAN_WAYPOINT)
+      {
+        auto cart_wp = std::static_pointer_cast<const tesseract_motion_planners::CartesianWaypoint>(waypoint);
+        plotAxis(cart_wp->getTransform(), 0.05);
+        if (cart_wp->getParentLinkName() != "world")
```

Comment:
This should be `if (!cart_wp->getParentLinkName().empty() && cart_wp->getParentLinkName() != env_->getSceneGraph()->getRoot())` . Also you could get the current state in this condition and update the published transform so it is in the correct frame.